src/imitation/algorithms/dagger.py

```diff
@@ -44,7 +44,7 @@ def __call__(self, round_num: int) -> float:
 class LinearBetaSchedule(BetaSchedule):
     """Linearly-decreasing schedule for beta."""
 
-    def __init__(self, rampdown_rounds: int):
+    def __init__(self, rampdown_rounds: int) -> None:
```

Comment:
https://mypy.readthedocs.io/en/stable/class_basics.html#annotating-init-methods I think it is an error to not return None. However, omitting it would not cause an error to be thrown.

This has not been added everywhere. In fact, there are a large number of missing annotations. I started to go through them, but stopped because this PR is in principle just to fix the errors that were raised by enabling mypy on these files, and I didn't want to add lots of extra noise to the changes. However, I kept the changes in the commits I had already pushed.

Reply:
Adding `-> None` to `__init__` seems like it should be doable with a `sed` one-liner. But I agree keeping this PR minimal is good, if we have another PR that just does that change it'll be easy to review because all the changes will be the same...

Having `__init__` annotated with None feels kinda silly to me: what else is it going to return? But according to https://stackoverflow.com/a/73334605/14507224 `mypy` will complain if there is an `__init__` method with no type annotations (it's OK if one of the arguments is typed though), so there's some argument for adding it everywhere for consistency.

Reply:
OK I change my mind, adding `-> None` to `__init__` does seem good given https://peps.python.org/pep-0484/ says:
> (Note that the return type of __init__ ought to be annotated with -> None. The reason for this is subtle. If __init__ assumed a return annotation of -> None, would that mean that an argument-less, un-annotated __init__ method should still be type-checked? Rather than leaving this ambiguous or introducing an exception to the exception, we simply say that __init__ ought to have a return annotation; the default behavior is thus the same as for other methods.)